packages/core/src/interpreter.ts

```diff
@@ -164,7 +164,7 @@ export class Interpreter<
   public status: InterpreterStatus = InterpreterStatus.NotStarted;
 
   // Actor
-  public parent?: Interpreter<any>;
+  public parent?: PrivateInterpreter<any>;
```

Comment:
I will have to change most of the `PrivateInterpreter` occurrences to `Interpreter`. It's a little bit problematic as `this as Interpreter` isn't quite OK for TS 😢 So the most sensible solution is to just use `as any` but usually I try to only keep that for special occasions and with this change I will have to do that in a lot of places here.